dataland-internal-storage/src/main/kotlin/db/migration/V19__MigrateNoEvidenceFoundAndIncomplete.kt

```diff
@@ -0,0 +1,63 @@
+package db.migration
+
+import db.migration.utils.DataTableEntity
+import db.migration.utils.getOrJavaNull
+import db.migration.utils.migrateCompanyAssociatedDataOfDatatype
+import org.flywaydb.core.api.migration.BaseJavaMigration
+import org.flywaydb.core.api.migration.Context
+import org.json.JSONObject
+
+/**
+ * This migration script moves all SFDR Incomplete quality buckets to NoDataFound and removes the value NoEvidenceFound
+ * by putting NoDataFound into the quality bucket and setting the value to null
+ */
+class V19__MigrateNoEvidenceFoundAndIncomplete : BaseJavaMigration() {
+
+    /**
+     * Find all data points with a value of NoEvidenceFound and move it to the quality bucket or with quality Incomplete
+     * and move it to NoDataFound
+     */
+    private fun recursivelyChangeRelevantDataPointFields(
+        dataset: JSONObject,
+        objectName: String,
```

Comment:
Ich denke `dataset` sollte lieber `object` heißen, denn es ist nur im Erst-Aufruf das gesamte dataset.
In den tieferen, rekursiven sub-Aufrufen ist es nur ein sub-Objekt des gesamten datasets.

`objectName` vllt eher `objectKey`.  Hier hab ich aber keine starke Meinung.

Reply:
geändert